txt.py and acq.py reorganization into io.py

vinferrer opened the pull request:
<!-- Write all of the issues that are linked to this pull request. -->
<!-- If this PR is enough to close them you can write something like "Closes #314 and closes #42" -->
<!-- If you just want to reference them without closing them, you can add something like "References #112" -->
Closes #295 

<!-- Add a short description of the PR content here-->
Merges `txt.py`and `acq.py` into `io.py`

## Proposed Changes
<!-- List major points of changes here, so that the reviewers can have a bit more context while looking at your work! -->

* Move the relevant functions to the same submodule (`phys2bids.io`). If you're concerned about their respective imports, you can just move the tool-specific imports into the function definitions.
* Rename the functions from `populate_phys_info()` to  `load_acq_ext()`, and `load_text_ext()` (or similar).
* Update tests and API documentation.

## Change Type
<!-- Indicate the type of change you think your pull request is -->
- [ ] `bugfix` (+0.0.1)
- [ ] `minor` (+0.1.0)
- [ ] `major`  (+1.0.0)
- [ ] `internal` (no version update)
- [ ] `documentation` (no version update)

## Checklist before review
<!-- If this section is not clear, please read this part of the docs: https://phys2bids.readthedocs.io/en/latest/contributorfile.html#pr -->
<!-- You're invited to open a draft PR ASAP, but before marking it "ready for review", check that you have done the following: -->
- [x] I added everything I wanted to add to this PR.
- [x] \[Code or tests only\] I wrote/updated the necessary docstrings.
- [x] \[Code or tests only\] I ran and passed tests locally.
- [ ] \[Documentation only\] I built the docs locally.
- [x] My contribution is harmonious with the rest of the code: I'm not introducing repetitions.
- [x] My code respects the adopted style, especially linting conventions.
- [x] The title of this PR is explanatory on its own, enough to be understood as part of a changelog.
- [x] I added or indicated the right labels.
<!-- If relevant, you can add a milestone label or indicate an ideal timeline for this PR, including whether the progress of this PR is linked to other PRs being completed before or after it. -->
- [ ] I added information regarding the timeline of completion for this PR.
<!-- If you want, you can ask for reviews on a draft PR -->
- [x] Please, comment on my PR while it's a draft and give me feedback on the development!

eurunuela commented:
Thank you @vinferrer ! I don't have the bandwidth right now to review this but I will. 

I see you selected both `minormod` and `majormod`. How's that?

Also, we would be going under 90% code coverage, so we'll need some tests.

vinferrer commented:
> Thank you @vinferrer ! I don't have the bandwidth right now to review this but I will.
> 
> I see you selected both `minormod` and `majormod`. How's that?
I don't know what to do about it.
> Also, we would be going under 90% code coverage, so we'll need some tests.
That's weird i didn't change any tests i just moved the functions around and modified the imports in the testing

vinferrer commented:
I forgot to add the testing function, :laughing:

eurunuela commented:
Now that I think of it, I believe this is an internal update. Not a `majormod` or a `minormod`. It really doesn't change the output of the package but makes our lives easier.

What do you think @tsalo @smoia ?

smoia commented:
Hey y'all! Sorry, I noticed I didn't add the refactoring to the possible change types. But @vinferrer labelled it right: it's a refactoring.

"Internal" is for the infrastructure. Maybe we can change the label, now that I think about it...

smoia commented:
BTW: I would get rid of the "interfaces" folder and rename `io.py` into `interfaces.py`. (It's not really an output function, is it?)

tsalo commented:
An interface implies a class to me, with bidirectional interaction with the source. Having an input/output module also lets you put the `write_json`, `write_file`, etc. functions in the same place.

eurunuela commented:
I like `io.py` better too.

vinferrer commented:
so should i get rid of the folder then ?

eurunuela commented:
> so should i get rid of the folder then ?

I would say yes. It does not make much sense to have the folder with just one file. And usually, `io.py` goes on the same level as `utils.py` for instance.

vinferrer commented:
okay

vinferrer commented:
I think this is ready for review then

smoia reviewed (commented):
Generally LGTM, but unless everyone else is ok with it, could you remove `_ext` from the name of the functions?

vinferrer commented:
> Generally LGTM, but unless everyone else is ok with it, could you remove `_ext` from the name of the functions?

Actually no, this is something that i put to remark that we are processing `extensions`  `.txt` is an extension, then we have formats like `acq` that can be in `.txt` or in `.acq` extensions

vinferrer commented:
This is actually related to the next PR I talk about in #295. Because my goal is to have extension functions that will call in the end to the same process fucntion. This way we will reduce the amount of code

smoia reviewed (approved):
Alright, sounds good!
It's ready!

eurunuela commented on phys2bids/io.py:
```diff
@@ -1,18 +1,21 @@
 
-"""phys2bids interface for txt files."""
-
+"""phys2bids interface for acqknowledge files."""
```
Why do you say it's the interface for acqknowledge files? We read txt files too.

vinferrer replied:
resolved

eurunuela reviewed (approved):
LGTM! I just have a tiny comment.

smoia commented:
@vinferrer , you merged a PR without being the main reviewer (not a big deal), but also without passing the style check.
Please open a PR now to pass it before anyone else opens a new PR.

smoia commented:
<!-- GITHUB_RELEASE COMMENT: released -->
:rocket: PR was released in [`2.3.0`](https://github.com/physiopy/phys2bids/releases/tag/2.3.0) :rocket:
